tests/templates/test_subroutines/test_select_pauli_rot.py

```diff
@@ -130,15 +130,37 @@ def test_correctness(self, angles, rot_axis):
 
         assert np.isclose(1.0, output[0])
 
-    def test_decomposition(self):
+    @pytest.mark.parametrize("n", [1, 2, 3, 4])
+    @pytest.mark.parametrize("axis", "XYZ")
+    @pytest.mark.parametrize("batch_dim", [None, 1, 3])
+    def test_decomposition(self, n, axis, batch_dim):
```

Comment:
Refined this test a little to include other rotation axis, qubit counts and broadcasting. Strictly speaking only the last addition tests code that has been changed in this PR.